### PR TITLE
Fix load uses compacted segments' binlogs

### PR DESCRIPTION
**Pull request by sunby**
Signed-off-by: sunby <bingyi.sun@zilliz.com>

/kind bug

### CODE REVIEW

File: internal/datanode/compactor.go [hunk 1]
```diff
@@ -458,9 +467,99 @@ func (t *compactionTask) compact() (*datapb.CompactionResult, error) {
...
+			insertLog.LogPath = blobPath
+		}
+
+		for _, deltaLog := range pack.deltaLogs {
+			splits := strings.Split(deltaLog.LogPath, "/")
+			if len(splits) < 1 {
+				pack.err = fmt.Errorf("delta stats log path: %s", deltaLog.LogPath)
+				return
+			}
+			logID, err := strconv.ParseInt(splits[len(splits)-1], 10, 64)
```
Comment (jaime0815) [inline internal/datanode/compactor.go:509]: use delta.GetLogID() to replace with parse log path

File: internal/datacoord/meta.go [hunk 2]
```diff
@@ -970,7 +979,29 @@ func (m *meta) PrepareCompleteCompactionMutation(compactionLogs []*datapb.Compac
...
+	return oldSegments, modSegments, segment, nil
+}
+
+func (m *meta) copyDeltaFiles(binlogs []*datapb.FieldBinlog, collectionID, partitionID, targetSegmentID int64) ([]*datapb.FieldBinlog, error) {
+	ret := make([]*datapb.FieldBinlog, 0, len(binlogs))
+	for _, fieldBinlog := range binlogs {
+		fieldBinlog = proto.Clone(fieldBinlog).(*datapb.FieldBinlog)
+		for _, binlog := range fieldBinlog.Binlogs {
+			blobKey := metautil.JoinIDPath(collectionID, partitionID, targetSegmentID, binlog.LogID)
+			blobPath := path.Join(m.chunkManager.RootPath(), common.SegmentDeltaLogPath, blobKey)
```
Comment (jaime0815) [inline internal/datacoord/meta.go:991]: you can use metautil.BuildDeltaLogPath function to replace with 990-991 line

File: internal/datanode/compactor.go [hunk 3]
```diff
@@ -458,9 +467,99 @@ func (t *compactionTask) compact() (*datapb.CompactionResult, error) {
...
+		for _, statsLog := range pack.statsLogs {
+			splits := strings.Split(statsLog.LogPath, "/")
+			if len(splits) < 2 {
+				pack.err = fmt.Errorf("bad stats log path: %s", statsLog.LogPath)
+				return
+			}
+			logID, err := strconv.ParseInt(splits[len(splits)-1], 10, 64)
+			if err != nil {
+				pack.err = err
+				return
```
Comment (jaime0815) [inline internal/datanode/compactor.go:538]: We'd better implement a method that replaces with segment id of log path to make the code clear.

File: internal/datanode/compactor.go (hunk 1 above)
Comment (sunby) [reply internal/datanode/compactor.go:509]: > use delta.GetLogID() to replace with parse log path

datanode does not write LogID filed when generating the datapb.deltaLog so for safe purpose I use the logpath to parse logID.

File: internal/datanode/compactor.go (hunk 3 above)
Comment (sunby) [reply internal/datanode/compactor.go:538]: > 

Good suggestion~

**Pull request merged**